### PR TITLE
[release 4.14]  OCPBUGS-23982: nmea_status and clock_class metrics missing in 4.14 linuxptpdaemon

### PR DESCRIPTION
**Pull request by aneeshkp**
Add nmea_status and clock class metrics

### CODE REVIEW

**Review by jzding — APPROVED** [review]
/lgtm

**Review by jzding — APPROVED** [review]
/lgtm

**Review by jzding — APPROVED** [review]
/lgtm

File: pkg/daemon/metrics.go [hunk 1]
```diff
@@ -201,16 +188,16 @@ func updatePTPMetrics(from, process, iface string, ptpOffset, maxPtpOffset, freq
...
+	configName = strings.Replace(strings.Replace(messageTag, "]", "", 1), "[", "", 1)
 	if strings.Contains(output, " max ") {
 		ifaceName, ptpOffset, maxPtpOffset, frequencyAdjustment, delay := extractSummaryMetrics(configName, processName, output)
 		if ifaceName != "" {
 			if ifaceName == clockRealTime {
 				updatePTPMetrics(phc, processName, ifaceName, ptpOffset, maxPtpOffset, frequencyAdjustment, delay)
 			} else {
 				updatePTPMetrics(master, processName, ifaceName, ptpOffset, maxPtpOffset, frequencyAdjustment, delay)
-				masterOffsetSource.set(configName, processName)
+				masterOffsetSourceProfile[configName] = processName
```
Comment (aneeshkp) [inline pkg/daemon/metrics.go:200]: Should use setter ,  for mutex , why changed?

File: pkg/daemon/metrics.go [hunk 2]
```diff
@@ -226,7 +213,7 @@ func extractMetrics(messageTag string, processName string, ifaces []string, outp
 				offsetSource = phc
 			}
 			if offsetSource == master {
-				masterOffsetSource.set(configName, processName)
+				masterOffsetSourceProfile[configName] = processName
```
Comment (aneeshkp) [inline pkg/daemon/metrics.go:216]: same here this will throw concurrent error

**Pull request merged**